core/src/main/scala/org/apache/spark/sql/execution/oplog/impl/OpLogRdd.scala

```diff
@@ -528,6 +529,16 @@ class OpLogRdd(
   override def compute(split: Partition, context: TaskContext): Iterator[Row] = {
     logDebug(s"starting compute for partition ${split.index} of table $fqtnUpper")
     try {
+      val currentHost = SparkEnv.get.executorId
+      val removePattern = "(executor_).*(_)".r
+      var expectedHost  = bucketHostMap.getOrElse(split.index,"")
+      require(expectedHost.nonEmpty, s"No preferred host found.")
```

Comment:
Give more informative reasons for requirement. Like a hint to user to what went wrong and point him to corrective action.

Reply:
yes, will modify the statement.